hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientExecutionServiceImpl.java

```diff
@@ -128,35 +110,17 @@ public void execute(Runnable command) {
      * @param command
      * @param delay
      * @param unit
-     * @param executionCallback
      * @return scheduledFuture
      */
-    public ScheduledFuture<?> schedule(final Runnable command, long delay, TimeUnit unit,
-                                       final ExecutionCallback executionCallback) {
-        return scheduledExecutor.schedule(new Runnable() {
-            public void run() {
-                executeInternalSafely(command, executionCallback);
-            }
-        }, delay, unit);
-    }
-
     @Override
     public ScheduledFuture<?> schedule(final Runnable command, long delay, TimeUnit unit) {
-        return scheduledExecutor.schedule(new Runnable() {
-            public void run() {
-                executeInternalSafely(command, failureLoggingExecutionCallback);
-            }
-        }, delay, unit);
+        return internalExecutor.schedule(command, delay, unit);
     }
 
     @Override
     public ScheduledFuture<?> scheduleWithRepetition(final Runnable command, long initialDelay, long period, TimeUnit unit) {
         final Runnable decoratedCommand = new SkipOnConcurrentExecutionDecorator(command);
-        return scheduledExecutor.scheduleAtFixedRate(new Runnable() {
-            public void run() {
-                executeInternalSafely(decoratedCommand, failureLoggingExecutionCallback);
-            }
-        }, initialDelay, period, unit);
+        return internalExecutor.scheduleAtFixedRate(decoratedCommand, initialDelay, period, unit);
```

Comment:
yes. eventually I'd like to have a common scheduler with the same semantic on both client and server. Basic infrastructure should be shared. I know it's not always possible, but it should be the ultimate goal.